main.js

```diff
@@ -28,5 +42,23 @@ define(function (require, exports, module) {
         // Register nohost server with highest priority
         LiveDevServerManager.registerServer({ create: _createServer }, 9001);
     });
-    
+
+        // XXXnote: This block of code will need to live in a Launcher of some
+        //          kind. It appears like the Brackets team were wanting to
+        //          add multiple launchers, which would make sense for multiple
+        //          transports. We might want to consider patching LiveDevMultBrowser
+        //          to allow setting the launcher as well as the transport, and adding
+        //          a launcher of our own rather than hacking on theirs.
+
+        // // send instrumented response or null to fallback to static file
+        // if (liveDocument && liveDocument.getResponseData) {
+        //     response = liveDocument.getResponseData();
+        // } else {
+        //     response = {};  // let server fall back on loading file off disk
+        // }
+        // response.id = request.id;
+
+        // this._send(request.location, response);
```

Comment:
We should be careful not to tangle the launcher with the liveDocument stuff.  Probably it needs to be something like this (this code is totally untested!):

```js
// In Launcher.js, assuming the `url` arg is really a raw pathname (might need NoHostServer to do this)
function launch(url) {
  var server = somehowGetReferenceToNoHostServer();
  var browser = somehowGetReferenceToIFrameBrowser();

  // Get the path from the NoHostServer, either as LiveDoc or from fs
  server.serveLiveDoc(url, function(err, url) {
    if(err) // somehow deal with err...
    browser.browse(url);
  });
}

// In NoHostServer.js, add a way to serve live docs vs. just files from fs
NoHostServer.prototype.serveLiveDoc = function(path, callback) {
  var fs = this.fs;
  var liveDocument = this._liveDocuments[path],

  function toURL(err, html) {
    if(err) // handle error somehow...
    // Convert rewritten HTML to a Blob URL Object
    var url = Content.toURL(html, 'text/html');
    callback(null, url); 
  }

  // If we have a LiveDoc for this path, send instrumented response. Otherwise fallback to static file from fs
  if (liveDocument && liveDocument.getResponseData) {
    Rewriter.rewriteHTML(liveDocument.getResponseData(), path, fs, toURL);
  } else {
    fs.readFile(path, 'utf8', toURL);
  }
};
```